google/cloud/aiplatform/datasets/tabular_dataset.py

```diff
@@ -163,15 +165,51 @@ def _retrieve_gcs_source_columns(
         finally:
             logger.removeFilter(logging_warning_filter)
 
-        return next(csv_reader)
+        return Set(next(csv_reader))
+
+    @staticmethod
+    def _get_bq_schema_field_names_recursively(schema_field: SchemaField) -> Set[str]:
+        """Retrieve the name for a schema field along with ancestor fields.
+        Nested schema fields are flattened and concatenated with a ".".
+        Schema fields with child fields are not included, but the children are.
+
+        Args:
+            project (str):
+                Required. Project to initiate the BigQuery client with.
+            bq_table_uri (str):
+                Required. A URI to a BigQuery table.
+                Can include "bq://" prefix but not required.
+            credentials (auth_credentials.Credentials):
+                Credentials to use with BQ Client.
+
+        Returns:
+            Set[str]
+                A set of columns names in the BigQuery table.
+        """
+
+        ancestor_names = [
+            nested_field_name
+            for field in schema_field.fields
+            for nested_field_name in TabularDataset._get_bq_schema_field_names_recursively(
+                field
+            )
+        ]
+
+        # Only return "leaf nodes", basically any field that doesn't have children
+        if len(ancestor_names) == 0:
+            return {schema_field.name}
+        else:
+            return {f"{schema_field.name}.{name}" for name in ancestor_names}
```

Comment:
Similar to the above comment, why do we treat ancestor names as a list but then convert the sequence to a set when  concatenating or returning the names?

Reply:
They can be sets too. I can change it.